examples/evaluate.py

```diff
@@ -0,0 +1,253 @@
+        path (str): Path to the directory with predictions. Can be a URL.
+        output_path (str): Output directory
+        dataset_path (str): Path to the dataset directory
+
+    Returns:
+        Metrics as a dictionary with metrics as the keys and metric values as the values
+    """
+
+    def get_replicates(dataset_name: str) -> List[Union[str, int]]:
+        if dataset_name == "camelyon17":
+            return list(range(0, 10))
+        elif dataset_name == "poverty":
+            return ["A", "B", "C", "D", "E"]
+        else:
+            return list(range(0, 3))
+
+    def get_best_prediction_filename(
+        dataset_name: str, split: str, replicate: Union[str, int]
+    ) -> str:
+        if dataset_name == "poverty":
+            return f"{dataset_name}_split:{split}_fold:{replicate}_epoch:best_pred.csv"
+        else:
+            return f"{dataset_name}_split:{split}_seed:{replicate}_epoch:best_pred.csv"
+
+    def get_metrics(dataset_name: str) -> List[str]:
+        if "amazon" == dataset_name:
+            return ["10th_percentile_acc", "acc_avg"]
+        elif "camelyon17" == dataset_name:
+            return ["acc_avg"]
+        elif "civilcomments" == dataset_name:
+            return ["acc_wg", "acc_avg"]
+        elif "fmow" == dataset_name:
+            return ["acc_worst_region", "acc_avg"]
+        elif "iwildcam" == dataset_name:
+            return ["F1-macro_all", "acc_avg"]
+        elif "ogb-molpcba" == dataset_name:
+            return ["ap"]
+        elif "poverty" == dataset_name:
+            return ["r_wg", "r_all"]
+        elif "py150" == dataset_name:
+            return ["acc", "Acc (Overall)"]
+        else:
+            raise ValueError(f"Invalid dataset: {dataset_name}")
+
+    # Dataset will only be downloaded if it does not exist
+    wilds_dataset: WILDSDataset = get_dataset(
+        dataset=dataset_name, root_dir=dataset_path, download=True
+    )
+    splits: List[str] = wilds_dataset.split_dict.keys()
+    if "train" in splits:
+        splits.remove("train")
+
+    replicates_results: Dict[str, Dict[str, List[float]]] = dict()
+    replicates: List[Union[str, int]] = get_replicates(dataset_name)
+    metrics: List[str] = get_metrics(dataset_name)
+
+    # Store the results for each replicate
+    for split in splits:
+        replicates_results[split] = {}
+        for metric in metrics:
+            replicates_results[split][metric] = []
+
+        for replicate in replicates:
+            predictions_file = get_best_prediction_filename(
+                dataset_name, split, replicate
+            )
+            print(
+                f"Processing split={split}, replicate={replicate}, predictions_file={predictions_file}..."
+            )
+            full_path = os.path.join(path, predictions_file)
+            predicted_labels: List[Any] = get_predictions(full_path)
+            predicted_labels_tensor: torch.Tensor = torch.from_numpy(
```

Comment:
Just as a heads up, we'll need to relax this when we add the detection datasets (because the predicted labels won't be a single Tensor), but we can deal with that later.